### PR TITLE
Fix generation of bignum literals

### PR DESCRIPTION
**Pull request by asandroq**
My previous attempt added an impossible condition that output all
bignum literals using the `MKBIGC` macro even when the value would fit
into a C int. This change fixes that bug.

### CODE REVIEW

File: src/IRTS/CodegenC.hs [hunk 1]
```diff
@@ -179,8 +179,10 @@ bcc i (ASSIGNCONST l c)
     = indent i ++ creg l ++ " = " ++ mkConst c ++ ";\n"
   where
     mkConst (I i) = "MKINT(" ++ show i ++ ")"
-    mkConst (BI i) | i > ((-2)^30) && i < (2^30) = "MKINT(" ++ show i ++ ")"
-                   | otherwise = "MKBIGC(vm,\"" ++ show i ++ "\")"
+    mkConst (BI i) = let maxInt = 2^30
```
Comment (ahmadsalim) [inline src/IRTS/CodegenC.hs]: On C at least the maximum integer is `INT_MAX` which is equal to 2^31-1.
The minimum integer is `INT_MIN` which is equal to -2^31.

see https://www.tutorialspoint.com/c_standard_library/limits_h.htm

Comment (ahmadsalim) [reply src/IRTS/CodegenC.hs]: Well for 32-bit integers anyway.

Comment (asandroq) [reply src/IRTS/CodegenC.hs]: That's true, but the last bit of an int is used to tag it as an unboxed value in the runtime.

Comment (ahmadsalim) [reply src/IRTS/CodegenC.hs]: Ah, I see.

Comment (Melvar) [reply src/IRTS/CodegenC.hs]: Well, since the size is based on a pointer, it’s either 31 or 63 bits depending on platform.

**Comment by melted** [comment]
It's slightly irritating that we can't reliably know if a long is 32 or 64 bits at codegen time. But big literals are rare, so it's way better to go with 32 bits than to hack something that could fail.

File: src/IRTS/CodegenC.hs [hunk 2]
```diff
@@ -179,8 +179,10 @@ bcc i (ASSIGNCONST l c)
...
+                     in if i >= -maxInt && i < maxInt
```
Comment (melted) [inline src/IRTS/CodegenC.hs]: You could use `if i < abs 2^30` for simplicity

Comment (ahmadsalim) [reply src/IRTS/CodegenC.hs]: You mean `abs i < abs 2^30`

Comment (asandroq) [reply src/IRTS/CodegenC.hs]: The test is different on either side of the interval.

Comment (melted) [reply src/IRTS/CodegenC.hs]: No I put the abs on the wrong side, 2^30 don't need an abs, it's a constant :-)

Comment (ahmadsalim) [reply src/IRTS/CodegenC.hs]: or `abs i < 2^30`

Comment (clayrat) [reply src/IRTS/CodegenC.hs]: spoiler alert: `x^30 = abs (x^30)` for any real `x` :D

**Comment by ahmadsalim** [comment]
Thanks

**Pull request merged**